### PR TITLE
RecordParser: Limit the record size

### PR DESCRIPTION
**Pull request by richardf**
Implements #2560.
If the record is too long, the RecordParser's exception handler will be called. After detecting a record whose size exceeds the maxRecordSize, the implementation will discard any data until a delimiter is found.

### CODE REVIEW

File: src/main/java/io/vertx/core/parsetools/impl/RecordParserImpl.java [hunk 1]
```diff
@@ -152,12 +155,33 @@ public void fixedSizeMode(int size) {
...
 
+  /**
+   * Set the maximum allowed size for a record when using the delimited mode.
+   * The delimiter itself does not count for the record size.
+   * <p>
+   * If a record is longer than specified, a RecordTooLongException will be thrown.
+   *
+   * @param size the maximum record size
+   */
+  public void maxRecordSize(int size) {
```
Comment (vietj) [inline src/main/java/io/vertx/core/parsetools/impl/RecordParserImpl.java]: make it fluent

**Comment by vietj** [comment]
I think we should do a simpler implementation, i.e instead of checking the boundary in `handleParsing ` and `parseDelimited` method, we should simply check that the remaining buffer size is lesser than the max size after `handleParsing` is called in the `handle` method.

**Comment by richardf** [comment]
@vietj The downside of this approach is that it will be difficult to resume the data ingestion after detecting a record that exceeds maxRecordSize. For my use case it is ok since I'll just close the socket and let the client reconnect. Wanna go that route?

**Comment by richardf** [comment]
@vietj updated as discussed.
The ip-validation failed because I committed from a different machine with other git configs. How do I fix that?

**Comment by vietj** [comment]
@richardf make a new clean PR from a clean commit

**Comment by richardf** [comment]
done. created #2635

**Comment by vietj** [comment]
thanks

**Pull request closed without merging**